lib/teleterm/daemon/daemon.go

```diff
@@ -553,20 +620,36 @@ func (s *Service) TransferFile(ctx context.Context, request *api.FileTransferReq
 	return cluster.TransferFile(ctx, request, sendProgress)
 }
 
+// notifyApp sends a notification (usually an error) to the Electron App.
+func (s *Service) notifyApp(ctx context.Context, notification *api.SendNotificationRequest) error {
+	tshdEventsCtx, cancelTshdEventsCtx := context.WithTimeout(ctx, tshdEventsTimeout)
+	defer cancelTshdEventsCtx()
+
+	_, err := s.tshdEventsClient.SendNotification(tshdEventsCtx, notification)
+	return trace.Wrap(err)
+}
+
 // Service is the daemon service
 type Service struct {
 	cfg *Config
 	// mu guards gateways and the creation of tshdEventsClient.
 	mu sync.RWMutex
+
 	// closeContext is canceled when Service is getting stopped. It is used as a context for the calls
 	// to the tshd events gRPC client.
 	closeContext context.Context
 	cancel       context.CancelFunc
 	// gateways holds the long-running gateways for resources on different clusters. So far it's been
 	// used mostly for database gateways but it has potential to be used for app access as well.
 	gateways map[string]*gateway.Gateway
+	// tshdEventsClient is a client to send events to the Electron App. This client is usually initialized
+	// during the creation of gateways, meaning it will not exist when the service is first initialized.
```

Comment:
I don't think that's correct, at least given the current state in this PR. `tshdEventsClient` is initialized during the call to `UpdateTshdEventsServerAddress` which is not related to gateways. It's called when the Electron app is starting up.